### ISSUE TITLE
Standards: map legends - syntax and rounding requirements

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#3653]
# problem

The map legend standards are a bit confusing.
In https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md it states 

> Map legend classes will have labels. The label should complement the legend title and should use the following syntax:
> 
> <label> <threshold> <count> <unit>
> 
> Label, refers to the class in the table above, i.e. affected, not affected, high hazard zone etc. threshold is the dynamic value which defines the class boundary i.e. the value of 0.7m for people affected by tsunami. count is the number of exposure elements in each class. Counts will be included in the legend label in brackets. These values will be rounded according to the legend label rounding principles. In some cases count will include a unit. The table below lists the units for each exposure type.

However the examples do not show this. The examples do not have rounded values and they do not include the threshold.

# proposed solution
- [ ] confirm that legends will alway have threshold or only if hazard is continuous raster data
- [ ] Update standards to show examples of legends

@felix-yew  - can you please review the issue about including thresholds and prepare some new examples
I will then update the standards document to reflect requirements.

# cc 
@Gustry @timlinux @samnawi

**Comment by Gustry** [comment]
Thanks @Charlotte-Morgan I will wait on this before to continue on implementing the standards about map legends.

**Comment by felix-yew** [comment]
Hi, @Charlotte-Morgan - As I remember when we discussed with the minion team in a meeting a couple months ago, there is an adjustment for the tsunami threshold into 0.7 m that comes from Pak Hamzah as tsunami expert. See the table below:
![image](https://cloud.githubusercontent.com/assets/13621886/22240090/0a24748c-e24c-11e6-8762-ffed899e60d1.png)

But, in the new InaSAFE version, the analysis will be polygonized. So, it will impact on the threshold and the map legend. For the example is tsunami hazard that will be affected, because previously it divided into two analysis, in population and the other (structures, roads, and land cover) become one analysis. 
Previously, in the tsunami raster with population raster analysis default set by 0.7 m and the impact result present by the total population in a hazard area. And then, the other (structures, roads, and land cover) the analysis default by 1 m, 3 m, and 8 m. Now, there is an innovation in the new architecture, all of the tsunami analysis with population, structures, roads, and land cover (~ CMIIW ~); analysis default set by:
- Dry zone < 0.1 m
- Low > 0.1 - 1 m
- Medium > 1 - 3 m
- High > 3 - 8 m
- Very high > 8 m

This is the example adapted to the new InaSAFE analysis result:
- People affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22240594/696b4018-e24e-11e6-836a-3e05bfafeff9.png)

- Buildings affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22241025/34cd3b34-e250-11e6-95f5-f144efbaf9fb.png)
In the number of population and buildings in the result not use unit parameter, just number.

- Roads affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22240539/3cc3f2da-e24e-11e6-95a0-f1a0ab1cabdf.png)

- Land cover affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22241106/90ca8914-e250-11e6-862d-79875eb679f0.png)

**Comment by Gustry** [comment]
I have some questions : 
- [ ] Do we need to round every values ? (not only population, but also roads, buildings) CF https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md#legend-label-rounding
According to the document, we need to apply the rounding to every kind of exposure. Is-it correct ?

- [ ] Do we need to add a new row in the impact summary about the `not exposed` ? https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md#examples-1
I think we are missing `not exposed` because Affected + Not affected + Not exposed = Total. 
<img width="857" alt="screen shot 2017-01-24 at 09 51 07" src="https://cloud.githubusercontent.com/assets/1609292/22240287/b6ceec44-e21a-11e6-9e54-f7a259cac0a5.png">

- [ ] Can you include the 'Not exposed' in : 
 - the table with the correct colour : https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md#examples-of-map-legend-titles
 - Example of map legends : https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md#examples-of-map-legends

By the way, this is a low priority, but the table https://github.com/inasafe/inasafe/blob/develop/docs/reporting-standards.md#examples-of-map-legend-titles is missing cyclone. @mbernasocchi added a new classification with a new color (very dark red)

**Comment by Charlotte-Morgan** [comment]
hi @felix-yew - looks like you may have misunderstood the requirement for this ticket.
Your response is very detailed but misses the fact that we are aware that InaSAFE 4.0 needs exposure specific thresholds for each hazard #3560 #3561 #3563 
Its also worth noting that polygonising things does not change thresholds.

There are 2 simple questions to address in this ticket:
- [ ] will legends always include thresholds or only if the hazard data are continuous rasters
- [ ] please produce new sample legends which include thresholds (if relevant) and rounding

**Comment by Charlotte-Morgan** [comment]
hi @Gustry : the standards say we will apply rounding to all exposures and to make it easy I used the same standard for building etc as for people. So yes, we do need rounding.

Is it possible for you to take all the other questions about tables to a separate ticket and assign it to me? - I would like this ticket to stay focussed on the legend only

**Comment by Gustry** [comment]
Yes sorry I will do.

@lucernae So we need to do the rounding to every kind of exposures.

**Comment by Charlotte-Morgan** [comment]
I think so - although it has only been population in the past. Your call if its possible to do all or only population now

**Comment by timlinux** [comment]
+1 from me to apply uniformly to all exposure types. We just need to make sure that the notes about rounding get added to all reports....

**Comment by Charlotte-Morgan** [comment]
hi @felix-yew - as discussed, we agree that all the syntax for legends in the standards is correct and that we would like all legends to include a threshold.  We believe this is possible for all hazards regardless of whether the hazard is continuous raster or classified vectors. The most difficult cases will be generic hazard and volcano buffers if the buffer distance is not included in the hazard data.

@felix-yew will provide some new samples for legends showing the syntax and rounding applied.
We can then update the standards doc with some new examples

**Comment by Charlotte-Morgan** [comment]
please note: we are only discussing syntax and rounding requirements for map legends in this ticket.

**Comment by felix-yew** [comment]
Hi, @Charlotte-Morgan - thank you for your explanation, this is some examples of map legends:
- People affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22276825/0adb8f8a-e2e9-11e6-91b4-0a069b32cb16.png)

- People affected by volcanic ash event
![image](https://cloud.githubusercontent.com/assets/13621886/22277143/d44409f4-e2eb-11e6-84b3-330a582fde53.png)

- Buildings affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22276836/2351acac-e2e9-11e6-98db-e74f2437ee5c.png)

- Roads affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22276858/3b5d7646-e2e9-11e6-94f4-487abe3cc2af.png)

- Land cover affected by tsunami event
![image](https://cloud.githubusercontent.com/assets/13621886/22276865/4739575a-e2e9-11e6-8344-1be49a2aefbe.png)

**Comment by timlinux** [comment]
One case that may not fit well are flood wet /dry classes....Could you please include in your draft how we should deal with those? Same for volcano hazard zones and generic hazard classes. I guess we need to figure out a rule to omit the parentheses in these cases?

**Comment by Charlotte-Morgan** [comment]
hi @timlinux - i think there should always be a number or area of things affected?

**Comment by Charlotte-Morgan** [comment]
@felix-yew - this is looking more complete now that it has thresholds, rounding and units.
I think that some of these examples still do not match well with the approach described in the standard.
https://cloud.githubusercontent.com/assets/5743135/17542495/6b968a16-5ef3-11e6-8622-af1cab2e0a5f.png
According to this figure from the standards:
- [ ] dry zone should only be included in the legend for roads and building exposed to flood and tsunami hazard; not for people or landcover.
- [ ] very low hazard is a pale green colour

is it possible for you to update the examples please

**Comment by Gustry** [comment]
If we use a vector flood polygon for instance (wet/dry), there is no thresholds. The same for generic. What should we display so ?
* High Hazard Zone
* Medium Hazard Zone
* High Hazard Zone

For volcano polygons, I guess it's the distance @timlinux : 
* High Hazard Zone <= 1000m
* Medium Hazard Zone > 1000m - 2000m
* High Hazard Zone > 2000m
Is-it correct like this @felix-yew and @Charlotte-Morgan ?

**Comment by Charlotte-Morgan** [comment]
For wet dry there are no thresholds (I think we are changing the default to petajakarta)
For volcano it will be buffer distance
For generic I think its not possible to show a threshold

**Comment by timlinux** [comment]
Ok the tricky thing for buffer distances is actually knowing what they are - we would need to get @myarjunar to write that info as metadata or column data in the buffer generation tool.

**Comment by felix-yew** [comment]
Hi @Charlotte-Morgan - I have updated the examples.

@Gustry yes, just like what Charlotte said.

**Comment by Charlotte-Morgan** [comment]
wow @felix-yew - that looks great thank you :)

**Comment by Charlotte-Morgan** [comment]
@Gustry - I see this comment also in the standards.  
**Note: thresholds are not included in the legend label for multiple event hazard data.** 
This means that thresholds are only included for single scenario hazard data; ie for hazard events not hazard zones

**Comment by Charlotte-Morgan** [comment]
@Gustry - a few points to note. Please pay attention to capitalisation; legend order from high to low; and no dry zone for people or landcover exposure.

**Comment by Gustry** [comment]
This is my current status in my work.
![screen shot 2017-01-25 at 10 41 22](https://cloud.githubusercontent.com/assets/1609292/22285164/de5964d0-e2ea-11e6-8764-99b6ac52dd30.png)

> This means that thresholds are only included for single scenario hazard data; ie for hazard events not hazard zones

I need to check this one.

> Please pay attention to capitalisation

The legend comes from the definitions, you can change the label there. https://github.com/inasafe/inasafe/blob/develop/safe/definitions/hazard_classifications.py

> no dry zone for people or landcover exposure

After the work from @ismailsunni if I understood, we will have a different hazard classification according to the exposure. So it will be in the wizard that you will not find dry zone for landcover.

**Comment by Charlotte-Morgan** [comment]
To make the label change do we need to change this string  from 'name': tr('Medium Hazard Zone'),
to 'name': tr('Medium hazard zone'),?

**Comment by Gustry** [comment]
> To make the label change do we need to change this string from 'name': tr('Medium Hazard Zone'),
to 'name': tr('Medium hazard zone'),?

Yes everything is driven by definitions in v4 ;-)

**Comment by Charlotte-Morgan** [comment]
ok - i have git mess right this minute - i need to fix my workflow before i can do more changes

**Comment by timlinux** [comment]
Hi

I was thinking about this some more:

@Charlotte-Morgan 

> For volcano it will be buffer distance

@timlinux 

> Ok the tricky thing for buffer distances is actually knowing what they are - we would need to get @myarjunar to write that info as metadata or column data in the buffer generation tool.


@Charlotte-Morgan are you assuming always one volcano in the hazard layer? If we have more than one volcano then I think it does not make sense to put the buffer distances in the legend - theoretically they might be different for each volcano.

**Comment by Charlotte-Morgan** [comment]
@timlinux - I was assuming and yes it would not be sensible for multiple volcanoes

**Comment by Gustry** [comment]
Does it mean the inasafe should detect if the hazard layer has many volcanoes ? And display the distance accordingly

**Comment by Charlotte-Morgan** [comment]
Would a single volcano be a single event and many volcanoes be multiple events?
We could use key words to make a rule on the behaviour?

**Comment by timlinux** [comment]
@Charlotte-Morgan I don't think that rule would work if the buffers represent KRB zones arent those based on multiple events? I guess using Etienne's approach could work but we still have the issue that we need to some how know that the hazard zones are based on buffers and what the buffer distances were -  I think my preference would be to not display buffer distances by computation. We could ask @myarjunar to write the buffer distance as part of the zone name e.g. ``KRB III 1000m`` - would that work?

**Comment by Charlotte-Morgan** [comment]
sure - lets try that.
